app/controllers/integrations/base_controller.rb

```diff
@@ -63,14 +67,18 @@ def find_or_create_release
     ReleaseService.new(project).release(release_params)
   end
 
-  # returns stage that failed to deploy or nil
+  # returns failed stage and any successfully started deploy ids
   def deploy_to_stages(release, stages)
     deploy_service = DeployService.new(user)
-    stages.detect do |stage|
+    stages.each_with_object(failed_stage: nil, deploy_ids: []) do |stage, result_hash|
```

Comment:
have 2 variables and build the hash at the end ?

Reply:
I kind of like the `each_with_object`, it feels a little cleaner to me than separately declaring two variables then explicitly returning the hash..